tensorflow/python/keras/layers/recurrent_v2.py

```diff
@@ -486,9 +486,12 @@ def step(cell_inputs, cell_states):
 def cudnn_gru(inputs, init_h, kernel, recurrent_kernel, bias, mask, time_major,
               go_backwards):
   """GRU with CuDNN implementation which is only available for GPU."""
-  if not time_major:
+  if not time_major and mask is None:
     inputs = array_ops.transpose(inputs, perm=(1, 0, 2))
-  init_h = array_ops.expand_dims(init_h, axis=0)
+    seq_axis, batch_axis = (0, 1)
+  else:
+    seq_axis, batch_axis = (0, 1) if time_major else (1, 0)
+  init_h = array_ops.expand_dims(init_h, axis=seq_axis)
```

Comment:
For the init_h here, it doesn't contain a sequence dim (it only contain one time step), should the expand_dim always apply to 0?

Reply:
No, it doesn't. The input_h shape is [num_layers, batch, num_units] for time_major, or [batch, num_layers, num_units] for batch_major. The expanded dim is num_layer which has the same position of seq dim.

Reply:
The input init_h here is always [batch, num_units] (since keras only handles single layer), if the cudnnrnn ops expect the different shape of input for init_h based on the time/batch major, probably add a comment here since the init state has nothing to do with timestep (it is for timestep 0).

Reply:
Yes, that makes sense. I will add some comments on this. Thx.